docs/topics/blocklist.md

```diff
@@ -33,109 +36,148 @@ If any changes are needed to the contents of the v1/v2 blocklist it must be made
-  "attachment_type": "bloomfilter-base",
-  "generation_time": 1587990908999,
-}
+      "schema": 1740062666739,
+      "attachment": {
+        "hash": "37ba24caec49afe6c97c424623e226e31ad052286ffa66d794eb82497dabc279",
+        "size": 28561,
+        "filename": "filter.bin",
+        "location": "staging/addons-bloomfilters/1234567890.bin",
+        "mimetype": "application/octet-stream"
+      },
+      "key_format": "{guid}:{version}",
+      "attachment_type": "bloomfilter-base",
+      "generation_time": 1587990908999,
+      "id": "851e162f-1227-486f-9abe-19bfdcb9fd12",
+      "last_modified": 1740063264110
+    }
 ```
 
+_(A similar record exists with `attachment_type: "softblocks-bloomfilter-base"` for the soft block base filter)_
+
 ### Stashing
 
 (blocklist-doc-stashing)=
 
-Because the bloomfilter files can be quite large "stash" files are also generated, which represent the changes since the previous bloomfilter generation and can be used by Firefox instead to save on bandwidth.
+Because the bloomfilter files can be quite large, for the **v3 blocklist**, "stash" files are also generated when changes are small compared to the base filters. These represent the changes (for **both hard and soft blocks**) since the _previous_ bloomfilter generation and can be used by Firefox v3 clients instead of downloading a full new filter, saving on bandwidth.
+
+Multiple stashes can be applied by Firefox (in chronological order using `stash_time`) to match the state of an up-to-date bloomfilter.
+
+The v3 stash data includes:
+
+- `blocked`: New hard blocks since the previous generation.
+- `unblocked`: Items no longer hard or soft blocked since the previous generation.
+- `soft_blocked`: New soft blocks since the previous generation.
+
+**Important:** When new base filters are uploaded (due to significant changes compared to the _base_, a missing base, or `force_base` being used), _all existing stash records are deleted_ from the Remote Settings collection. Clients fetching updates after this point will need to download the new base filters, and stashing effectively restarts for v3 clients.
 
-Multiple stashes can be applied by Firefox (in chronological order) to match the state of an up-to-date bloomfilter.
+Stash data is stored directly within the Remote Settings record JSON, not as an attachment.
 
-#### Stash record example
+#### Stash record example (v3)
 
 ```json
 {
+  "id": "example-stash-record-67890",
+  "last_modified": 1678887000000,
+  "key_format": "{guid}:{version}",
+  "stash_time": 1678887000123, // Example Timestamp (milliseconds)
   "stash": {
     "blocked": [
-      "{6f6b1eaa-bb69-4cdb-a24f-1014493d4290}:10.48",
-      "kittens@pioneer.mozilla.com:1.2",
-      "kittens@pioneer.mozilla.com:1.1",
-      "{b01e0601-eddc-4306-886b-8a4fb5c38a1e}:1",
-      "{232f11df-20ca-49d4-94eb-e3e63d7ae773}:1.1.2",
-      "kittens@pioneer.mozilla.com:1.3",
+      "{guid1}:{versionA}",
+      "{guid2}:{versionB}"
```

Comment:
The examples with real guids and versions are better than these placeholder - can we have them back?  We can slightly modify the guids if we're concerned with exposing real block data (which shouldn't be a concern anyway - it's publicly available data)